symex-ts.el

```diff
@@ -0,0 +1,261 @@
+            child
+          (symex-ts--descend-to-child-with-sibling child))
+      node)))
+
+(defun symex-ts--ascend-to-parent-with-sibling (node)
+  "Ascend from NODE to parent recursively.
+
+Recursion will end when the parent node has a sibling or is the
+root."
+  (let ((parent (tsc-get-parent node)))
+    (if parent
+        (if (or (tsc-get-prev-named-sibling parent) (tsc-get-next-named-sibling parent))
+            parent
+          (symex-ts--ascend-to-parent-with-sibling parent))
+      node)))
+
+(defun symex-ts--move-with-count (fn move-delta &optional count)
+  "Move the point from the current node if possible.
+
+Movement is defined by FN, which should be a function which
+returns the appropriate neighbour node.
+
+MOVE-DELTA is a two-element list describing the desired x and y
+point movement.
+
+Move COUNT times, defaulting to 1.
+
+Return a Symex move (list with x,y node offsets tagged with
+'move) or nil if no movement was performed."
+  (let ((target-node nil)
+        (move symex--move-zero)
+        (cursor (symex-ts-get-current-node)))
+    (dotimes (_ (or count 1))
+      (let ((new-node (funcall fn cursor)))
+        (when (and new-node (not (eq new-node cursor)))
+          (setq move (symex--add-moves
+                      (list move
+                            (symex-make-move (car move-delta) (cadr move-delta)))))
+          (setq cursor new-node
+                target-node cursor))))
+    (when target-node (symex-ts--set-current-node target-node))
+
+    ;; Return the Symex move that was executed, or nil to signify that
+    ;; the movement failed
+    (when (not (symex--are-moves-equal-p move symex--move-zero)) move)))
+
+(defun symex-ts--after-tree-modification ()
+  "Handle any tree modification."
+  (symex-ts--delete-overlay)
+  (setq-local symex-ts--current-node nil))
+
+
+(defun symex-ts-current-node-sexp ()
+  "Print the current node as an s-expression."
+  (interactive)
+  (message (tsc-node-to-sexp symex-ts--current-node)))
+
+(defun symex-ts-get-current-node ()
+  "Return the current node.
+Automatically set it to the node at point if necessary."
+  (unless symex-ts--current-node
+    (symex-ts-set-current-node-from-point))
+  symex-ts--current-node)
+
+(defun symex-ts-set-current-node-from-point ()
+  "Set the current node to the top-most node at point."
+  (symex-ts--set-current-node (symex-ts-get-topmost-node-at-point)))
+
+(defun symex-ts-get-topmost-node-at-point ()
+  "Return the top-most node at the current point."
+  (let ((root (tsc-root-node tree-sitter-tree))
+        (p (point)))
+    (symex-ts--get-topmost-node (tsc-get-named-descendant-for-position-range root p p))))
+
+
+(defun symex-ts-move-prev-sibling (&optional count)
+  "Move the point to the current node's previous sibling if possible.
+
+Move COUNT times, defaulting to 1."
+  (interactive "p")
+  (symex-ts--move-with-count #'tsc-get-prev-named-sibling '(-1 0) count))
```

Comment:
Any reason we couldn't pass `(symex-make-move -1 0)` here? That would avoid `car` and `cadr` in `symex-ts--move-with-count`

Reply:
Good catch! I've just pushed an update for this.